financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/dcr/application/listener/FSApplicationManagementListener.java

```diff
@@ -0,0 +1,184 @@
+            throws IdentityApplicationManagementException {
+
+        try {
+            AssociatedRolesConfig rolesConfig = new AssociatedRolesConfig();
+            rolesConfig.setAllowedAudience("ORGANIZATION");
+            serviceProvider.setAssociatedRolesConfig(rolesConfig);
+
+            LocalAndOutboundAuthenticationConfig localAndOutboundAuthenticationConfig = serviceProvider
+                    .getLocalAndOutBoundAuthenticationConfig();
+
+            identityDataHolder.getAbstractApplicationUpdater()
+                    .doPreCreateApplication(serviceProvider, localAndOutboundAuthenticationConfig, tenantDomain,
+                            userName);
+
+        } catch (FinancialServicesException e) {
+            log.error("Error occurred while Creating application.", e);
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean doPostCreateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+            throws IdentityApplicationManagementException {
+
+        try {
+            APIResourceManager resourceManager = IdentityExtensionsDataHolder.getInstance().getApiResourceManager();
+            APIResource apiResource = resourceManager.getAPIResourceByIdentifier("User-defined-oauth2-resource",
+                    tenantDomain);
+            if (apiResource == null) {
+                return false;
+            }
+
+            AuthorizedAPI authorizedAPI = new AuthorizedAPI();
+            authorizedAPI.setAPIId(apiResource.getId());
+            authorizedAPI.setScopes(apiResource.getScopes());
+            authorizedAPI.setPolicyId("RBAC");
+
+            IdentityExtensionsDataHolder.getInstance().getAuthorizedAPIManagementService()
+                    .addAuthorizedAPI(serviceProvider.getApplicationResourceId(), authorizedAPI, tenantDomain);
+
+            identityDataHolder.getAbstractApplicationUpdater().doPostCreateApplication(serviceProvider,
+                    serviceProvider.getLocalAndOutBoundAuthenticationConfig(), tenantDomain, userName);
+
+        } catch (APIResourceMgtException e) {
+            log.error("Error occurred while retrieving API resource.", e);
+            return false;
+        } catch (IdentityApplicationManagementException e) {
+            log.error("Error occurred while adding authorized API.", e);
+        } catch (FinancialServicesException e) {
+            log.error("Error occurred while creating application.", e);
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean doPreUpdateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+            throws IdentityApplicationManagementException {
+
+        try {
+            OAuthConsumerAppDTO oAuthConsumerAppDTO = DCRUtils
+                    .getOAuthConsumerAppDTO(serviceProvider.getApplicationName());
+            oAuthConsumerAppDTO.setTokenType("JWT");
```

Comment:
there was a recent fix done to this to be able to set a custom token type, shall we include that as well? @VimukthiRajapaksha did the fix for OB3 AFAIR

Reply:
This is the one, 
https://github.com/wso2/financial-services-accelerator/pull/313

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/306/commits/ebbb3de3520013ca48f27c23e66ff50dcdf43170